### ISSUE TITLE
Redundant operation in parseMessage

### ISSUE THREAD

**Issue by akolotov** [poanetwork/poa-bridge-contracts#211]
During the security audit the following comment was raised:

> `and(, 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)` is redundant as this check is done automatically as recipient variable has address type

https://github.com/poanetwork/poa-bridge-contracts/blob/bbb97a63c900f03a902d0e82358abac3b294e4d9/contracts/libraries/Message.sol#L57-L62

The investigation showed that this line was added as part of work under the previous audit comment: https://github.com/mixbytes/audits_public/blob/master/solidity/POANetwork/audit_en.md#2messagesol-27

From another point of view, the following experiment shows that the check is done automatically indeed:
![image](https://user-images.githubusercontent.com/20793260/60278145-86fc1900-9907-11e9-9d6d-f069c82ca4b7.png)

Consider to perform more experiments to check if the discussed code is redundant or not.

**Comment by patitonar** [comment]
Used https://remix.ethereum.org to run the tests and the following contract:

```solidity
pragma solidity 0.4.24;

contract TestContract {
        function parseMessage(bytes message) public pure returns(address recipient, address withoutCheck)
    {
        assembly {
            recipient := and(mload(add(message, 20)), 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
            withoutCheck := mload(add(message, 20))
        }
    }
}
```

Input 
```
parseMessage(0xce42bdb34189a93c55de250e011c68faee374dd3)
```

Output
```
{
	"0": "address: recipient 0xce42bdB34189a93c55De250E011c68FaeE374Dd3",
	"1": "address: withoutCheck 0xce42bdB34189a93c55De250E011c68FaeE374Dd3"
}
```

Calldata
```
0x52af1c0400000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000014ce42bdb34189a93c55de250e011c68faee374dd3000000000000000000000000
```

Looking at debug steps for `recipient`:
- `mload(add(message, 20))` result is `0x000000000000000000000014ce42bdb34189a93c55de250e011c68faee374dd3`
- `and(mload(..), 0xFFF..)` result is `0x000000000000000000000000ce42bdb34189a93c55de250e011c68faee374dd3`
- Value asigned to solidity local variable `recipient` is `0xCE42BDB34189A93C55DE250E011C68FAEE374DD3`

Looking at debug steps for `withoutCheck`:
- `mload(add(message, 20))` result is `0x000000000000000000000014ce42bdb34189a93c55de250e011c68faee374dd3`
- Value asigned to solidity local variable `withoutCheck` is `0xCE42BDB34189A93C55DE250E011C68FAEE374DD3`

From previous auditor's recomendation the mask `0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF` was added to avoid relying on zeros that may or may not be there as mentioned [here](https://github.com/mixbytes/audits_public/blob/master/solidity/POANetwork/audit_en.md#2messagesol-27).

The security audit recommended to remove the check since it is done automatically because of the variable type. From the experiments, we can see that other bytes read along with the address are not only zeros (`000000000000000000000014`). These other bytes are ignored and the correct value is assigned to the variables even in cases when the mask is not provided, so I think that the mask is not needed and it makes sense to remove it.

**Comment by akolotov** [comment]
Included in [TokenBridge contracts 2.3.3](https://github.com/poanetwork/poa-bridge-contracts/releases/tag/2.3.3)
